ably/realtime_conn.go

```diff
@@ -500,7 +500,11 @@ func (c *Connection) setSerial(serial int64) {
 	c.serial = serial
 }
 
-func (c *Connection) eventloop() {
+func (c *Connection) eventloop() (err error) {
+	defer func() {
+		c.forciblyNackPendingMessages(err)
```

Comment:
[RTN19](https://docs.ably.io/client-lib-development-guide/features/#RTN19)
>Any ProtocolMessage that is awaiting an ACK/NACK on the old transport will not receive the ACK/NACK on the new transport. The client library must therefore resend any ProtocolMessage that is awaiting a ACK/NACK to Ably in order to receive the expected ACK/NACK for that message. The Ably service is responsible for keeping track of messages, ignoring duplicates and responding with suitable ACK/NACK messages

If we nack them now, how will we be able to re send them? I got stuck a bit implementing `RTN19` thinking of a better way to test it.

Reply:
Hmm, then I'd say this part of RTN15f is in conflict with RTN19: "once a transport drops, the client library must either fail the publish attempt, or re-attempt by re-sending the messages on a new transport" RTN19 makes it non-optional to re-send the messages.

Also, this makes RTN15f redundant, so I'm closing the PR.

Will open a spec change with this.

Reply:
See https://github.com/ably/docs/pull/948.